packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4734,11 +4734,53 @@ class Listener extends SingleChildRenderObjectWidget {
 ///
 /// This widget creates a separate display list for its child, which
 /// can improve performance if the subtree repaints at different times than
-/// the surrounding parts of the tree. Specifically, when the child does not
-/// repaint but its parent does, we can re-use the display list we recorded
-/// previously. Similarly, when the child repaints but the surround tree does
-/// not, we can re-record its display list without re-recording the display list
+/// the surrounding parts of the tree.
+///
+/// This is useful since [RenderObject.paint] may be triggered even if its
+/// associated [Widget] instances' did not change or rebuild, such as when an
+/// ancestor scrolls or when an ancestor or descendent's widget is going through
+/// an animation.
```

Comment:
Maybe a better way to phrase this:

> A [RenderObject] will repaint whenever any [RenderObject] that shares the same [Layer] is marked as
> being dirty and needing paint (see [RenderObject.markNeedsPaint]). The [RepaintBoundary] widget
> creates a [RenderObject] that always has a [Layer], decoupling the ancestor render objects from the
> descendant render objects.
>
> For example, if a render object's rendering is being animated, its parent will typically also be repainted
> each frame. By placing a [RepaintBoundary] between the animated child and the parent, the parent
> is decoupled from the child and so will no longer repaint each frame of the animation.

Reply:
Added in your wording, though I spread them out a bit in various paragraphs so reading the whole thing sequentially doesn't interlace problem statements with solutions back and forth